htdocs/core/class/commonobject.class.php

```diff
@@ -1060,18 +1060,25 @@ public function liste_contact($statut = -1, $source = 'external', $list = 0, $co
 	{
         // phpcs:enable
 		global $langs;
+        
+        $langs->load("dict");
 
 		$tab=array();
 
 		$sql = "SELECT ec.rowid, ec.statut as statuslink, ec.fk_socpeople as id, ec.fk_c_type_contact";    // This field contains id of llx_socpeople or id of llx_user
 		if ($source == 'internal') $sql.=", '-1' as socid, t.statut as statuscontact, t.login, t.photo";
 		if ($source == 'external' || $source == 'thirdparty') $sql.=", t.fk_soc as socid, t.statut as statuscontact";
-		$sql.= ", t.civility as civility, t.lastname as lastname, t.firstname, t.email";
+		$sql.= ", t.civility as civility, t.lastname as lastname, t.firstname, t.address, t.zip, t.town, t.fk_pays as country_id, t.fk_departement as state_id";
+		$sql.= ", t.phone, t.phone_perso, t.phone_mobile, t.fax, t.email";
 		$sql.= ", tc.source, tc.element, tc.code, tc.libelle";
+        $sql.= ", co.label as country, co.code as country_code";
+		$sql.= ", d.nom as state, d.code_departement as state_code";
 		$sql.= " FROM ".MAIN_DB_PREFIX."c_type_contact tc";
 		$sql.= ", ".MAIN_DB_PREFIX."element_contact ec";
 		if ($source == 'internal') $sql.=" LEFT JOIN ".MAIN_DB_PREFIX."user t on ec.fk_socpeople = t.rowid";
 		if ($source == 'external'|| $source == 'thirdparty') $sql.=" LEFT JOIN ".MAIN_DB_PREFIX."socpeople t on ec.fk_socpeople = t.rowid";
+        $sql.= " LEFT JOIN ".MAIN_DB_PREFIX."c_country as co ON t.fk_pays = co.rowid";
+		$sql.= " LEFT JOIN ".MAIN_DB_PREFIX."c_departements as d ON t.fk_departement = d.rowid";
```

Comment:
If source is "internat", the table is llx_user and field fk_pays and fk_departement are fk_country and fk_state.

Reply:
@eldy do you want fix in this function or fix in contact DB & contact.class.php ?